public/app/features/scenes/dashboard/DashboardsLoader.ts

```diff
@@ -285,15 +280,24 @@ export function createVizPanelFromPanelModel(panel: PanelModel) {
       // To be replaced with it's own option persited option instead derived
       hoverHeader: !panel.title && !panel.timeFrom && !panel.timeShift,
       $data: createPanelDataProvider(panel),
+      menu: new VizPanelMenu({
+        $behaviors: [panelMenuBehavior],
```

Comment:
This is clever, but wonder what's the benefit you see here compared to this approch: https://github.com/grafana/scenes/blob/main/packages/scenes-app/src/demos/panelMenu.ts#L46 ? I would imagine building panel menu via a function - some static elements will always be there, and possibly handling dynamic items using a plain activation handler without behaviors.

Reply:
@dprokop 
it's the exact same thing, I prefer the behavior as I can define everything inline, it's more declarative. A stateless behavior is just an activation handler. 

The behavior could potentially be serializable (if we wanted to "snapshot" the runtime state of a dashboard), where an externally added activation handler could not be.